Arrow.swift

```diff
@@ -80,20 +103,14 @@ public func <== <T:ArrowParsable>(inout left:T?, right: AnyObject?) {
 // Suppport Array of custom Types
 
 public func <== <T:ArrowParsable>(inout left:[T], right: AnyObject?) {
-    left = [T]()
-    if let pns = right as? [AnyObject] {
-        for pn in pns {
-            left.append(T(json:pn))
-        }
+    if let a = right as? [AnyObject] {
```

Comment:
Here the `left` value gets overwritten even in the case when parsing failed.
Also I made the code a little more compact here :)